xblock/mixins.py

```diff
@@ -610,3 +611,19 @@ def has_support(self, view, functionality):
             True or False
         """
         return hasattr(view, "_supports") and functionality in view._supports  # pylint: disable=protected-access
+
+
+class CompletableXBlockMixin(object):
```

Comment:
This and the emit_completion method should have docstrings.  The class docstring should include a brief note regarding why it should be used. (For blocks that need special control over when they are considered complete).  The method docstring should explain that emitting a value of 1.0 indicates that a block has been completed, and that emitting a value of 0.0 on a previously completed block indicates that it is no longer considered complete.

Reply:
@jcdyer I don't think it should exit silently - this mixin doesn't have anything useful besides this method - so falling back to default should be performed by not mixin in this class.